### ISSUE TITLE
`performance-move-const-arg` not reported with `emplace_back()` and such

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#86447]
```cpp
#include <string>
#include <vector>

class C
{
public:
    C(int i, const std::string& s);
};

int main()
{
    std::vector<C> v;

    {
        std::string s = "str";
        
        C c(0, std::move(s));
    }
    

    {
        std::string s = "str";
        
        v.push_back(C(0, std::move(s)));
    }
    
    {
        std::string s = "str";
        
        v.emplace_back(0, std::move(s)); // no warning
    }
}
```

```
<source>:17:16: warning: passing result of std::move() as a const reference argument; no move will actually happen [performance-move-const-arg]
   17 |         C c(0, std::move(s));
      |                ^~~~~~~~~~ ~
<source>:24:26: warning: passing result of std::move() as a const reference argument; no move will actually happen [performance-move-const-arg]
   24 |         v.push_back(C(0, std::move(s)));
      |                          ^~~~~~~~~~ ~
```
https://godbolt.org/z/vqM1orqdP

**Comment by PiotrZSL** [comment]
Yes, this is because emplace_back is an perfect forwarding template.
Same issue would be if you would construct C with std::make_unique.
It's not easy fix, but probably for some methods/functions support could be added, but not sure if it's worth it.
